### DISCUSSION TITLE
Best 2.0-forwards-compatible practices when using pgbouncer?

### DISCUSSION THREAD

**Discussion by saolof** [sqlalchemy/sqlalchemy#8751] [Usage Questions]
Using sqlalchemy to connect to pbouncer can lead to some headaches particularly when your user has a capped number of db connections which can cause different processes to deadlock while requesting connections when server active connections are stuck in an "idle in transaction" mode, which can happen even for selects if you don't have autocommit on. 

Some standard practices (always fill out appname, use nullpool or limit pool size to 1-2 connection, enable autocommit) can be found in this post: https://dba.stackexchange.com/questions/36828/how-to-best-use-connection-pooling-in-sqlalchemy-for-pgbouncer-transaction-level/118105#118105

So I was wondering what the best practices are for 2.0? Since previously session autocommit was the most straightforward way to ensure that simple selects did not lead to idle transactions.

**Comment by CaselIT** [comment]
Hi,

simple select do not lead to "idle in transaction" if the application properly manage the connection, ie closes the session/connection when it is done with it.
As already replied in gitter, you can use the dbapi autocommit https://docs.sqlalchemy.org/en/20/core/connections.html#dbapi-autocommit
